corehq/apps/geospatial/models.py

```diff
@@ -85,16 +85,16 @@ def disbursement_solver(self):
         ]
 
     @property
-    def api_token(self):
-        if self._api_token and self._api_token.startswith(f'${ALGO_AES}$'):
-            ciphertext = self._api_token.split('$', 2)[2]
+    def plaintext_api_token(self):
+        if self.api_token and self.api_token.startswith(f'${ALGO_AES}$'):
+            ciphertext = self.api_token.split('$', 2)[2]
             return b64_aes_decrypt(ciphertext)
-        return self._api_token
+        return self.api_token
 
-    @api_token.setter
-    def api_token(self, value):
+    @plaintext_api_token.setter
+    def plaintext_api_token(self, value):
         if value and not value.startswith(f'${ALGO_AES}$'):
             ciphertext = b64_aes_encrypt(value)
-            self._api_token = f'${ALGO_AES}${ciphertext}'
+            self.api_token = f'${ALGO_AES}${ciphertext}'
         else:
-            self._api_token = None
+            self.api_token = None
```

Comment:
nit: I am thinking that this else block is implicitly resetting the api_token.

I would think that this function would encrypt and store the value and if it is not able to should it
1. do nothing
2. raise an error

I would prefer 2. More specifically:
```
if value is None:
    self.api_token = None
elif value and not value.startswith(f'${ALGO_AES}$'):
    # encrypt and store
else:
    raise Exception("Unexpected value received for plaintext api token")
```

Let me know what you think.

Reply:
https://github.com/dimagi/commcare-hq/pull/34024/commits/335f4c805ee18af13a41257b632b77af93d05b58

The setter now also handles the case where you set a non-string value, say, `1234`. You also can't set an empty api_token. The token can only ever be either
1. A string value (not empty)
2. null